src/ClimaTimeSteppers.jl

```diff
@@ -57,6 +57,16 @@ export AbstractAlgorithmName, AbstractAlgorithmConstraint, Unconstrained, SSP
 array_device(::Union{Array, SArray, MArray}) = CPU()
 array_device(x) = CUDADevice() # assume CUDA
 
+function similar_zero(x)
+    s = similar(x)
+    if hasmethod(fill!, typeof(x))
+        fill!(s, 0)
+    else
+        @warn "Please ensure that all tendencies are initialized or set in tendency functions given to ClimaODEFunction."
```

Comment:
Would this warning be triggered in ClimaAtmos?

Reply:
Hmm, yes, actually.

Reply:
I suppose we could just apply the `(Yₜ, u, _, t) -> (Yₜ .= 0);` change, but what I've added is at least an increment safer than what is in the main branch.

Reply:
If I understand this correctly, this warning would be triggered also for valid code. What should users do to prevent it?

Reply:
> What should users do to prevent it?

Ensure that this object passed to the timestepper supports `fill!`, but I'm now more leaning towards changing this to require `zero` be defined. ClimaCore probably just should have supported `Base.zero` for these data structures from the start, like they do with `Base.similar`.

Now that I think about it, this is not technically a breaking change from the point of view from CTS, since it doesn't depend on ClimaCore.

I'm inclined to make a new release in ClimaCore, and then update this PR to just use `Base.zero`

Reply:
Ok, I've updated the PR to just use `Base.zero` instead of `Base.similar`.